core/IO.carp

```diff
@@ -90,7 +92,11 @@ module are wrappers around the C standard library.")
   (doc      get-line "gets a line from stdin.")
   (register get-line (Fn [] String))
   (doc      fgetc "gets a character from a file pointer (thin wrapper for the C standard library).")
-  (register fgetc (Fn [(Ptr FILE)] Char)) ; TODO: check EOF handling (see carp_io.h)!
+  (defn fgetc [file]
+    (let [char (IO.Raw.fgetc file)]
+      (if (or (IO.Raw.feof file) (IO.Raw.ferror file))
```

Comment:
Should those two errors be distinct?

Reply:
yeah, they should--another thing I was wondering about is if we should define an actual Error type for these errors rather than just a string.

Reply:
I’m usually too lazy for this, but it’s probably the correct thing to do :)

Reply:
added! -- eventually we'll want to check `errno` and report on what the error was, but for now we at least distinguish between EOF and other errors.